### DISCUSSION TITLE
Debugger and Profound UI

### DISCUSSION THREAD

**Discussion by bigbearjeff63** [codefori/vscode-ibmi#2131] [IBM i] [answered]
Hi there. We use Profound UI for our web front end Applications and aren't sure how to associate the debugger with our RPGLE Code via VS Code. These are web-based Applications so there's nothing "called" from a green screen. Any chance you might be able to guide us on how to make the debugger in VSCODE run while working with a Profound Web Screen?

**Comment by sebjulliand** [comment, marked as answer]
Perfect timing! The Debug Service v2 has been released this week by IBM and it adds support for service entry points.
If you have the IBM i Debug extension, it's probably updated already.
Now you need to install the PTF that will bump the debug service to V2:
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/7ca7fc5d-a6c3-498c-af84-4444fb324ddd)

You'll also need Java 11 on the LPAR you're going to debug.

Once it's updated, Code for IBM i will detect you're using the V2 and enable the `Set Service Entry Point` action the debug menu (when you right click on a `.PGM` or `.SRVPGM` object in the Object Browser).

Once you have set an entry, as soon as the program/service program gets called on the LPAR, VSCode will pick it up and start a debug session.

**Reply by bigbearjeff63** [reply]
You're Amazing! Thank you!

**Reply by sebjulliand** [reply]
You're welcome!
Also, you may need to update the debug service configuration for V2, so it uses Java 11. See my comment here: https://github.com/codefori/vscode-ibmi/issues/2114#issuecomment-2180651302

**Reply by bigbearjeff63** [reply]
Silly Follow up Question. Can you tell me where all of my compiles are going now? In case I wanted to look up something I compiled earlier today?

**Reply by sebjulliand** [reply]
If you use the standard actions, compiles are going in the current library defined in the library list view.

**Reply by bigbearjeff63** [reply]
In green screen world I could typically use the WRKSPLF command to see all of my compiles. So, they're not landing there. If you're referring to the library list that's setup under the User Library List. I'm not sure how to discern where from that list things might be landing. Can you be more specific. I'm worried I'm missing something obvious here. Sorry to be a pain.

**Reply by sebjulliand** [reply]
No worries! I thought you were referring to your compiled objects. So forget about the current library.

When compiling from Code for i, no spooled file gets generated. Everything goes into the action output. You can open the output right after the action has run, but I'm afraid it doesn't get saved anywhere like spooled files are.
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/9f1f1031-d6dc-4d16-ad09-95ee3da79561)
